packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -797,14 +771,44 @@ const VirtualListBaseFactory = (type) => {
 			delete rest.spacing;
 			delete rest.updateStatesAndBounds;
 
+			const
+				{dataSize, direction, itemSize, overhang, rtl, spacing} = this.prevProps,
+				{props} = this,
+				hasMetricsChanged = (
+					direction !== props.direction ||
+					((itemSize instanceof Object) ? (itemSize.minWidth !== props.itemSize.minWidth || itemSize.minHeight !== props.itemSize.minHeight) : itemSize !== props.itemSize) ||
```

Comment:
nit: we can use ramda's `equals` function to simplify.

`!equals(prevProps.itemSize, this.props.itemSize)`

Reply:
Oh, it looks much better. Thanks.

Reply:
<On Feb 18th>
- I updated the PR.